master/internal/kubernetes/pod.go

```diff
@@ -101,9 +149,189 @@ func (p *pod) startPod(ctx *actor.Context) error {
 	}
 }
 
-func (p *pod) configureResourcesRequirements() v1.ResourceRequirements {
-	return v1.ResourceRequirements{
-		Limits: map[v1.ResourceName]resource.Quantity{
+func (p *pod) receivePodStatusUpdate(ctx *actor.Context, msg podStatusUpdate) error {
+	p.pod = msg.updatedPod
+
+	switch msg.updatedPod.Status.Phase {
+	case k8sV1.PodPending:
+		// When pods are deleted, Kubernetes sometimes transitions pod statuses
+		// to pending prior to deleting them. We ignore this state if we have
+		// triggered the deletion.
+		if p.resourcesDeleted {
+			ctx.Log().Info("ignoring pod status because resources are being deleted")
+			return nil
+		}
+
+		containerState := getContainerState(msg.updatedPod.Status.Conditions)
+		if containerState == container.Running {
+			ctx.Log().Errorf("unexpected containers status while pod is pending")
+		}
+
+		if containerState == p.container.State {
+			return nil
+		}
+
+		if containerState == container.Starting {
+			// Kubernetes does not have an explicit state for pulling container
+			// images. We insert it here because our  current implementation of
+			// the trial actor requires it.
+			ctx.Log().Infof("transitioning pod state from %s to %s",
+				p.container.State, container.Pulling)
+			p.container = p.container.Transition(container.Pulling)
+
+			rsc := sproto.ContainerStateChanged{Container: p.container}
+			ctx.Tell(p.taskHandler, rsc)
+		}
+
+		ctx.Log().Infof("transitioning pod state from %s to %s", p.container.State, containerState)
+		p.container = p.container.Transition(containerState)
+
+		rsc := sproto.ContainerStateChanged{Container: p.container}
+		ctx.Tell(p.taskHandler, rsc)
+
+	case k8sV1.PodRunning:
+		if p.container.State == container.Running {
+			return nil
+		}
+		p.container = p.container.Transition(container.Running)
+
+		logStreamer, err := newPodLogStreamer(p.podInterface, p.podName, ctx.Self())
+		if err != nil {
+			return err
+		}
+		_, ok := ctx.ActorOf(fmt.Sprintf("%s-logs", p.podName), logStreamer)
+		if !ok {
```

Comment:
```suggestion
		if _, ok := ctx.ActorOf(fmt.Sprintf("%s-logs", p.podName), logStreamer); !ok {
```

Reply:
oh yeah should made this change once we don't the need to track the logs actor anymore